staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4996,11 +4996,21 @@ type EndpointPort struct {
 	// +optional
 	Protocol Protocol `json:"protocol,omitempty" protobuf:"bytes,3,opt,name=protocol,casttype=Protocol"`
 
-	// The application protocol for this port.
+	// Used as a hint for implementations to
+	// configure the protocol used between the
+	// implementation and the application it exposes.
 	// This field follows standard Kubernetes label syntax.
-	// Un-prefixed names are reserved for IANA standard service names (as per
+	// Valid values are either:
+	//
+	// * Un-prefixed protocol names - reserved for IANA standard service names (as per
 	// RFC-6335 and https://www.iana.org/assignments/service-names).
-	// Non-standard protocols should use prefixed names such as
+	//
+	// * Kubernetes standard names:
+	//   * 'k8s.io/h2c' - http2 over cleartext as described in https://www.rfc-editor.org/rfc/rfc7540
+	//   * 'k8s.io/grpc' - grpc traffic - see https://github.com/grpc/grpc/blob/v1.51.1/doc/PROTOCOL-HTTP2.md
+	//   * 'k8s.io/tcp' - plain tcp traffic
```

Comment:
Why do we need this?  Isn't `protocol: TCP` plus the absence of `appProtocol` an indicator of "plain" TCP ?

Reply:
Apparently no. 
I think nowadays it is up to the implementation, For example, as far as i remember, in Istio, in case of`protocol: TCP` and  absence of `appProtocol`  they try to sniff and guess the protocol. However, with `appProtocol: TCP` they treat it as plain TCP.
@howardjohn is that right?

Reply:
Yep

Reply:
so we're asking people to specify "TCP" twice in order to defend against implementations?  That seems...unpleasant.

If we need to add a hint here, can we call it "raw" or something, which works for all port-protocols and tells implementations "don't try to be clever".